### DISCUSSION TITLE
Is MENU PASSWD settings still there?

### DISCUSSION THREAD

**Discussion by lgraziosi** [cobbler/cobbler#2861] [Version specific questions] [answered]
Hello, I need to put a simple password in front of the PXE boot loader, to not let the "dumb" users to wrongly launch an installation.  
I've looked up the documentation and it looks like there was this setting in an older version:
https://cobbler.readthedocs.io/en/release28/4_advanced/pxe%20boot%20menu%20passwords.html#references
Was it removed? It doesn't seems to work on UEFI boots at least, and I don't find it at all in the newer documentation.
Thank you.

**Comment by SchoolGuy** [comment, marked as answer]
It appears that is was removed indeed. However you can change the template again so it does require it.

For details please see: https://wiki.syslinux.org/wiki/index.php?title=Menu#MENU_PASSWD

**Reply by lgraziosi** [reply]
Thank you for your reply.
I've added it on those files and it seems to only work with BIOS PXE boot.
Should I change other files to make it work on EFI?
Thanks again.

**Reply by SchoolGuy** [reply]
I never tried that option myself but if you are using EFI, I believe you are using GRUB. Thus maybe the following document helps you to set a password for GRUB: https://help.ubuntu.com/community/Grub2/Passwords

P.S.: I am not sure if this document still is applying but if it is, it would be nice to know so I can link that in the docs.

**Reply by lgraziosi** [reply]
It could be possible I guess, I just need to understand how to create the configuration files for cobbler, and how to create them in a way that `cobbler sync` doesn't break anything.
Thank you, anyway.

**Reply by SchoolGuy** [reply]
The secret is to not edit the files in the TFTP-Boot-Directory directly but instead the templates from which Cobbler generates the data from.

**Reply by lgraziosi** [reply]
Yeah, the grub.cfg is easy to find because I see its original folder on the `cobbler sync` command, but the ` x86_64_menu_items.cfg` file seems to be generated at every launch, and I don't know where it takes its informations from.

**Reply by SchoolGuy** [reply]
The information is taken from the distros, profiles and systems you enter into Cobbler. The template which is used should be (if I am not mistaken) `grub_submenu.template`.

**Reply by lgraziosi** [reply]
It's `grubprofile.template` and it actually works!
Thanks.
